src/docs/asciidoc/reference/architecture/code-structure.adoc

```diff
@@ -1,2 +1,150 @@
+====
+
+[#role_tsconfig]
+=== tsconfig.json
+
+The `tsconfig.json` file is here because by default ZetaPush uses the `TypeScript ecosystem`. For example, the _cloud services_ are written in `TypeScript` and the `worker` folder contains by default a `TypeScript` file.
+You can see <<reference#javascript_vs_typescript, why we use TypeScript>>.
+
+In the `TypeScript ecosystem`, the `tsconfig.json` file is needed to define the root of the application. So in our case, the `tsconfig.json` file is placed at the root of the *ZetaPush application*.
+
+If you are not familiar with `TypeScript`, you can see more informations link:{typescript-url}[here].
+
+In addition to define the root of our application, the `tsconfig.json` file define how the transpilation from `TypeScript` to `JavaScript` is done.
+You can define a lot of configurations according to the link:{typescript-conf-url}[TypeScript documentation].
+
+For the proper functioning of the *ZetaPush application*, the `tsconfig.json` file needs to have some minimal configuration like explained below :
+
+include::{docdir}/common/code/minimal-tsconfig-needed.adoc[]
+
+.Customize my tsconfig.json
+
+In this case of you need to customize your `tsconfig.json` file, for example to use classes only declared in ES6 (es2015), you can add this in the `compilerOptions`. Of course, you can add any properties provided by link:{typescript-conf-url}[TypeScript].
+
+.Customized tsconfig.json
+[source,json,linenums]
+----
+{
+  "compilerOptions": {
+    "lib": [
+      "es2015"
+    ],
+    "experimentalDecorators": true,
+    "emitDecoratorMetadata": true
+  }
+}
+----
+
+=== worker folder
+
+As seen above, the `worker` folder contains the business logic of your application. It can also called your "back code".
+This is in this part that you will create your business logic and interact with the _built-in cloud services_ provided by ZetaPush (User management, file system, database, notifications, etc...).
+
+By default (When the application is generated by the _CLI_), the `worker` folder contains only one file, called `index.ts`.
+
+include::{docdir}/common/code/generated-worker-code.adoc[]
+
+.Develop your worker in several ways
+[NOTE]
+====
+In this example we see a worker that is created with only one class. You will see in the <<reference.adoc#develop_custom_cloud_services, cloud service development section>> how to create a worker in several ways.
+====
+
+.Don't need to transpile TypeScript
+[NOTE]
+====
+Even if your worker code is in `TypeScript` you don't need to transpile it to `JavaScript`. Indeed, when you run or push your code, it will be launched with link:{ts-node-url}[`ts-node`]. +
+`ts-node` take care of the transpilation, you can also configure this via the `tsconfig.json` file.
+====
+
+=== front folder
+
+The `front` folder contains your front code.
+So you need to have your `index.html` (or other html file if you want) at the root of this folder.
```

Comment:
Not mandatory at root even if it is better. It could be in folder front/toto/index.html and then the url would be http://localhost:3000/toto. The words "you need to have" feels that it won't work otherwise but developer can organize its code how he wants